### PR TITLE
feat: send docker image layers SHAs

### PR DESCRIPTION
**Pull request by orkamara**
- [X] Ready for review
- [X] Follows CONTRIBUTING rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
In order to collect improve our docker image matching, we want to start looking on the layers SHAs.
In his PR we'll start collect this info by extracting the layers SHAs from the output of `docker inspect` cmd.

### CODE REVIEW

File: test/lib/analyzer/image-inspector.test.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env node_modules/.bin/ts-node
+// Shebang is required, and file *has* to be executable: chmod +x file.test.js
+// See: https://github.com/tapjs/node-tap/issues/313#issuecomment-250067741
+
+// tslint:disable:max-line-length
+// tslint:disable:object-literal-key-quotes
```
Comment (robcresswell) [inline test/lib/analyzer/image-inspector.test.ts]: Don't disable linting please; lets just fix the errors, or disable for specific blocks, not whole files :smile:

File: test/lib/analyzer/index.test.ts [hunk 2]
```diff
@@ -77,12 +77,24 @@ test('analyzer', async t => {
...
 
-  const imageIdStub = sinon.stub(imageIdDetector, 'detect')
-    .resolves('sha256:fake');
+  const expectedId = 'sha256:fake';
+  const expectedLayers =
+  ['sha256:fake1',
+  'sha256:fake2',
+  'sha256:fake3'];
+
+  const stubbedData = { Id: expectedId,
```
Comment (robcresswell) [inline test/lib/analyzer/index.test.ts]: Can we fix the formatting here?

```
const stubbedData = {
  Id: expectedId,
  RootFS: {
    Layers: [] // inline them here, it doesn't seem like the tests are reusing those expected vars, so no need to abstract them
  }
```

etc.

File: lib/analyzer/image-inspector.ts [hunk 3]
```diff
@@ -4,10 +4,10 @@ export {
   detect,
 };
 
-async function detect(targetImage: string): Promise<string> {
+async function detect(targetImage: string): Promise<any> {
```
Comment (robcresswell) [inline lib/analyzer/image-inspector.ts]: Let's type this please, no `any` :)

File: test/lib/analyzer/image-inspector.test.ts (hunk 1 above)
Comment (orkamara) [reply test/lib/analyzer/image-inspector.test.ts]: didn't touch this code :)
It looks new because of the renaming...

But I'll fix it anyway

Comment (robcresswell) [reply test/lib/analyzer/image-inspector.test.ts]: Ah, couldn't tell. Leave it then, I thought it was new cause of the diff.

**Pull request merged**